File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.ts
function twoSum(numbers: number[], target: number): number[] {
  let possibleResult = new Map();

  for (let i = 0; i < numbers.length; i++) {
    const restNumber = target - numbers[i];
    if (possibleResult.has(restNumber)) {
      return [possibleResult.get(restNumber) +1, i+1];
    } else {
      possibleResult.set(numbers[i], i);
    }
  }

  return [];
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        start = 0;
        end = len(numbers) - 1
        
        
        while start < end:
            tem_sum = numbers[start] + numbers[end]
            if tem_sum == target:
                return [start+1, end+1]
            elif tem_sum < target:
                start += 1
            else:
                end -= 1
        
        return []